proposals/csharp-7.0/tuples.md

```diff
@@ -0,0 +1,437 @@
+
+```csharp
+var t = (sum: 0, count: 1);
+
+System.ValueTuple<int, int> vt = t;  // identity conversion
+(int moo, int boo) t2 = vt;          // identity conversion
+
+t2.moo = 1;
+```
+
+> note:
+>
+> An element name at one position on one side of a conversion, and the same name at another position on the other side almost certainly have bug in the code:
+
+```csharp
+(string first, string last) GetNames() { ... }
+(string last, string first) names = GetNames(); // Oops!
+```
+
+> Compilers should issue a warning for the preceding code.
+>
+> endnote.
+
+### Boxing conversions
+
+> The following text should be added to [Boxing conversions](../../spec/conversions.md#boxing-conversions) after the first paragraph:
+
+Tuples, like all value types, have a boxing conversion. Importantly, the names aren't part of the runtime representation of tuples, but are tracked only by the compiler. Thus, once you've "cast away" the names, you cannot recover them. In alignment with the identity conversions, a boxed tuple will unbox to any tuple type that has the same element types in the same order.
+
+### Tuple conversions
+
+> This section should be added after [Implicit enumeration conversions](../../spec/conversions.md#implicit-enumeration-conversions)
+
+Tuple types and expressions support a variety of conversions by "lifting" conversions of the elements into overall *tuple conversion*.
+For the classification purpose, all element conversions are considered recursively. For example: To have an implicit conversion, all element expressions/types must have implicit conversions to the corresponding element types.
+
+Tuple conversions are *Standard Conversions* and therefore can stack with user-defined operators to form user-defined conversions.
+
+An implicit tuple conversion is a standard conversion. It applies between two tuple types of equal arity when there is any implicit conversion between each corresponding pair of types.
+
+An explicit tuple conversion is a standard conversion. It applies between two tuple types of equal arity when there is any explicit conversion between each corresponding pair of types.
+
+A tuple conversion can be classified as a valid instance conversion or an extension method invocation as long as all element conversions are applicable as instance conversions.
+
+On top of the member-wise conversions implied by target typing, implicit conversions between tuple types themselves are allowed.
+
+### Target typing
+
+> This section should be added after [Anonymous function conversions and method group conversions](../../spec.md#Anonymous-function-conversions-and-method-group-conversions)
+
+A tuple literal is "target typed" when used in a context specifying a tuple type. The tuple literal has a "conversion from expression" to any tuple type, as long as the element expressions of the tuple literal have an implicit conversion to the element types of the tuple type.
```

Comment:
>type [](start = 148, length = 4)

of the same length

Reply:
Fixed in #3150. Used "arity" for consistency.